nasa/earthdata-search
# EDSC-2450: Implemented Datepicker and DatepickerContainer with the expected functionality

# Overview

### What is the feature?

Changed the Datepicker/DatepickerContainer so that when closing the calendar/input (blurring), dates autocomplete depending on the datetime format that was set in the input.

### What is the Solution?

Updated Datepicker, DatepickerContainer, and their tests (all associated tests that are impacted by the change)

### What areas of the application does this impact?

List impacted areas.

# Testing

### Reproduction steps

- **Environment for testing:**
- **Collection to test with:**

1. Go to localhost:8080/search
2. Open the temporal datetime filter.
3. Enter 1999 in the start & end inputs
4. observe the fields autocompleting to 1999-01-01 00:00:00 & 1999-12-31 23:59:59 respectively
5. Enter 1999-02 in start and end inputs
6. observe the fields autocompleting to 1999-02-01 00:00:00 & 1999-02-28 23:59:59 respectively
7. edit the start date to 1999-01-06 12:00:00 and end date to 1999-02-06 13:00:00
8. observe the dates don't autocomplete
9. select a date using the calendar for the start and end dates and observe they autocomplete correctly.
### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

We have good coverage on tests when run locally.
<img width="616" alt="Screenshot 2024-08-13 at 1 16 11 PM" src="https://github.com/user-attachments/assets/9f43d7a2-67fa-4617-9cba-e75c551e439a">
<img width="753" alt="Screenshot 2024-08-13 at 1 16 34 PM" src="https://github.com/user-attachments/assets/863c7179-04d8-4447-8c32-18e1c39c06b7">

# Checklist
